### PR TITLE
Revise mod

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - clean FusedOp legacy from mod operator
 - add type_prop tests
 - add backend tests

### Tickets:
 - 44644

### CODE REVIEW

File: ngraph/core/src/op/mod.cpp [hunk 1]
```diff
@@ -12,58 +12,70 @@
...
-
-constexpr NodeTypeInfo op::v1::Mod::type_info;
-
-op::v1::Mod::Mod()
-    : FusedOp()
-    , m_auto_broadcast()
+namespace modop
 {
+    template <element::Type_t ET>
+    bool evaluate(const HostTensorPtr& arg0,
```
Comment (jdanieck) [inline ngraph/core/src/op/mod.cpp:23]: @bszmelcz was addition of evaluate() for mod requested from someone? Originally ref impl was called from evaluates_map() so based on that I think we shouldn't add it. However, if it was explicitly requested then we need to remove it from evaluates_map.

File: ngraph/test/backend/mod.in.cpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    auto f = make_shared<Function>(T2, ParameterVector{A, B});
+
+    vector<float> a{1, 2, 3, 4};
+    vector<float> b{5, 6, 7, 8};
+
+    auto test_case = test::TestCase<TestEngine>(f);
+    test_case.add_multiple_inputs<float>({a, b});
+    test_case.add_expected_output<float>(shape, {0, 0 ,0 ,0});
+    test_case.run();
+}
```
Comment (jdanieck) [inline ngraph/test/backend/mod.in.cpp:133]: Please add tests with negative numbers.

**Review by jdanieck — CHANGES_REQUESTED** [review]

File: ngraph/core/src/op/mod.cpp (hunk 1 above)
Comment (bszmelcz) [reply ngraph/core/src/op/mod.cpp:23]: Oh yes yes, I forgot to remove it, since I already asked Gabri about it

**Review by jdanieck — APPROVED** [review]

File: ngraph/test/type_prop/mod.cpp [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#include "arithmetic_ops.hpp"
```
Comment (lazarevevgeny) [inline ngraph/test/type_prop/mod.cpp:1]: Copyright header?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**